[ENTSWM-554] restclient works using CompletionStage and @PathParam

penehyba opened the pull request.

Ladicek commented on microprofile/microprofile-rest-client-1.3/src/main/java/org/wildfly/swarm/ts/microprofile/rest/client/v13/ClientAutoCloseable.java:
```diff
@@ -1,13 +1,19 @@
 package org.wildfly.swarm.ts.microprofile.rest.client.v13;
 
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
```
No. Use the JAX-RS types.

penehyba commented:
Few changes:
import org.jboss.resteasy.annotations.jaxrs.PathParam;
->
import javax.ws.rs.PathParam;
moved test from microprofile-rest-client-1.3 to microprofile-rest-client-1.2 (easier to simulate this issue)
fixed typo
